examples/simpleGov/README.md

```diff
@@ -0,0 +1,1064 @@
+    Option      string          // Option chosen by voter
+    Voter       sdk.Address     // Address of the voter
+}
+```
+
+## Implementation
+
+Now, that we have our types defined, we can start actually implementing the module.
+
+First, let us go into the module's folder and create a folder for our module.
+
+```bash
+cd x/
+mkdir simple_governance
+cd simple_governance
+```
+
+Let us start by adding the files we will need. Your module's folder should look something like that:
+
+```
+x
+└─── simple_governance
+      ├─── client
+      │     ├───  cli
+      │     │     └─── simple_governance.go
+      │     └─── rest
+      │           └─── simple_governance.go
+      ├─── errors.go
+      ├─── handler_test.go
+      ├─── handler.go
+      ├─── keeper_keys.go
+      ├─── keeper_test.go
+      ├─── keeper.go
+      ├─── test_common.go
+      ├─── types_test.go
+      ├─── types.go
+      └─── wire.go
+```
+
+Let us go into the detail of each of these files.
+
+### Types
+
+In this file, we define the custom types for our module. This includes the types from the [State](#State) section and the custom message types defined in the [Transactions](#Transactions) section.
+
+For each new type that is not a message, it is possible to add methods that make sense in the context of the application. In our case, we will implement an `updateTally` function to easily update the tally of a given proposal as vote messages come in.
+
+Messages are a bit different. They implement the `Message` interface defined in the SDK's `types` folder. Here are the methods you need to implement when you define a custom message type:
+
+- `Type()`: This function returns the name of our module's route. When messages are processed by the application, they are routed using the string returned by the `Type()` method.
+- `GetSignBytes()`: Returns the byte representation of the message. It is used to sign the message.
+- `GetSigners()`: Returns address(es) of the signer(s).
+- `ValidateBasic()`: This function is used to discard obviously invalid messages. It is called at the beginning of `runTx()` in the baseapp file. If `ValidateBasic()` does not return `nil`, the app stops running the transaction.
+- `Get()`: A basic getter, returns some property of the message.
+- `String()`: Returns a human-readable version of the message
+
+For our simple governance messages, this means:
+
+- `Type()` will return `"simpleGov"`
+- For `SubmitProposalMsg`, we need to make sure that the attributes are not empty and that the deposit is both valid and positive. Note that this is only basic validation, we will therefore not check in this method that the sender has sufficient funds to pay for the deposit
+- For `VoteMsg`, we check that the address and option are valid and that the proposalID is not negative.
+- As for other methods, less customization is required. You can check the code to see a standard way of implementing these.
+
+### Keeper
+
+#### Short intro to keepers
```

Comment:
Keepers Explanation?

Reply:
Also may want to link to file in the header of the section

Reply:
what's wrong with "Short intro to keepers"? :p